src/main/java/nextstep/subway/applicaion/LineService.java

```diff
@@ -77,4 +81,14 @@ public void deleteSection(Long lineId, Long stationId) {
 
         line.deleteSection(station);
     }
+
+    private Section createSection(Line line, Station upStation, Station downStation, int distance, int duration) {
+        return new Section.Builder()
+                .line(line)
+                .upStation(upStation)
+                .downStation(downStation)
+                .distance(distance)
+                .duration(duration)
+                .build();
+    }
```

Comment:
지금 매개변수가 5개나 되버렸네요. 😢 
중복이 되기에 메서드로 하려고하는건 좋지만, 매개변수가 오히려 늘어나 버렸네요 ㅠ

```suggestion
    private Section createSectionV2(Long lineId, SectionRequest request) {
        Station upStation = stationService.findById(request.getUpStationId());
        Station downStation = stationService.findById(request.getDownStationId());
        
        return new Section.Builder()
                .line(findById(lineId))
                .upStation(upStation)
                .downStation(downStation)
                .distance(request.getDistance())
                .duration(request.getDuration())
                .build();
    }
```
이런식으로 할 수는 없을까요? 여기에 몇가지 유효성검증이나 유연성만 키워서요 ㅎ

Reply:
아... 한솔님께서 어떤 의미로 피드백을 주신지 이해했습니다!

> 제대로 이해한지 모르겠지만 ㅎㅎ...

- 매개변수가 많아지면 `가독성`이 떨어진다
- 매개변수가 많아지면, 해당 메서드의 `책임`이 많을 수도 있다

따라서,

" `메서드 추출` 기법을 사용하여 리팩토링을 하더라도, 매개변수가 많아진다면 해당 메서드의 매개변수를 줄일 수 있는 방법을 생각해보는 것이 좋다"

이렇게 이해했습니다.

피드백을 참고하여 다시 리팩토링 해보겠습니다!

Reply:
현재 리팩토링 진행과정 중에 있습니다.

`addSection()` 이랑 `saveLine()` 두 곳에서 createSection 을 사용하는데, 한솔님께서 피드백 주신 코드의 형태로 작성하는게 힘들어 보입니다.

saveLine 메서드와 addSection 메서드의 DTO 매개변수가 서로 다릅니다. 

> saveLine : LineRequest, addSection : SectionRequest

그러다 보니 `메서드 추출` 기법을 쓰더라도,  매개변수를 줄이기가 힘들다는 생각이 드는데요.. 
이런 경우에는 SectionBuilder 를 사용하는 코드 자체가 중복이긴 하지만, 메서드 추출보다, 그냥 각 메서드에서 builder 를 사용하는게 `가독성` 측면에서 더 낫다고 보여집니다.

> 물론, 메서드 추출 기법을 써서 매개변수를 줄일 수 있을 것 같긴합니다.  그럴라면 LineRequest 와 SectionRequest 에서 CreateSectionDTO 같은 것을 만들어줘서 해당 객체로 변환한다음 아래 처럼 사용해야할 것 같습니다.

```java
private Section createSectionV2(Long lineId, CreateSectionRequest request) {
        Station upStation = stationService.findById(request.getUpStationId());
        Station downStation = stationService.findById(request.getDownStationId());
        
        return new Section.Builder()
                .line(findById(lineId))
                .upStation(upStation)
                .downStation(downStation)
                .distance(request.getDistance())
                .duration(request.getDuration())
                .build();
    }
```

정리하자면

1. 메서드 추출을 하지말고 builder 를 사용하여 Section 객체를 생성한다.
2. 메서드 추출 기법을 쓰기 위해서, 섹션 객체를 생성하기 위한 DTO 를 하나 만들어 준다.

크게 정답은 없는 것 같지만 굳이.. DTO 를 추가로 만들어줘야 하는 의문이 들긴해서, 1번 방법을 사용하여 리팩토링을 하고 피드백을 받아보겠습니다.